### PR TITLE
fix(reports): show all reports when favorites toggle is off

### PR DESCRIPTION
**Pull request by swatikothari**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |

Related issues: #502 ...

### CODE REVIEW

File: ui/src/features/get-last-reports.js [hunk 1]
```diff
@@ -43,16 +43,18 @@ class getReports extends React.Component {
     };
   }
 
-    filterByFavoriteState = () => {
+    filterFavorites = () => {
       const { onlyFavorites, sortedReports } = this.state;
-      const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
-      this.setState({ sortedReports: filteredReports, sortHeader: '' });
+      if (onlyFavorites) {
+        const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
```
Comment (enudler) [inline ui/src/features/get-last-reports.js:49]: can be changed to:
const filteredReports = sortedReports.filter((report) => (report.is_favorite));

File: ui/src/features/get-test-reports.js [hunk 2]
```diff
@@ -53,16 +53,18 @@ class getTests extends React.Component {
         this.setState({rerunJob: job});
     };
 
-    filterByFavoriteState = () => {
+    filterFavorites = () => {
         const {onlyFavorites, sortedReports} = this.state;
-        const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
-        this.setState({sortedReports: filteredReports, sortHeader: ''});
+        if (onlyFavorites) {
+            const filteredReports = sortedReports.filter((report) => (!!report.is_favorite) === onlyFavorites);
```
Comment (enudler) [inline ui/src/features/get-test-reports.js:59]: Can be changed to
const filteredReports = sortedReports.filter((report) => (report.is_favorite));

**Comment by enudler** [comment]
hi @swatikothari, thanks again!
I will accept this PR once you address the minor the comments

but the idea was to query the server with the filter instead of the filter in memory

**Comment by swatikothari** [comment]
Sure @enudler - I can do another PR for that referencing the same issue (or a new issue if that should be created).

**Comment by enudler** [comment]
Thanks @swatikothari same issue is great

**Pull request merged**